wezterm-gui/src/line-frag.glsl

```diff
@@ -23,9 +23,9 @@ void main() {
   vec4 under_color = sample_texture(atlas_nearest_sampler, o_underline);
   if (under_color.a != 0.0) {
     // if the underline glyph isn't transparent in this position then
-    // we take the underline color, otherwise we'll leave the color
+    // we overlay the underline color, otherwise we'll leave the color
     // at the background color.
-    color = o_underline_color;
+    color = color * (1.0 - under_color) + under_color * o_underline_color;
```

Comment:
My original intent for this was to draw the underline color over the top of the text if the underline was set, but the implementation was a bit wonky as you've discovered.

Can you share a screenshot of how this looks?  It may not be strictly required; for example, an alternative to this is `color = under_color` to take all of the rgba information from your computed underline glyph.

Reply:
The reasoning was, that if the Underline is drawn over the cursor, the half transparent pixels should be rendered over the cursor. I'm not sure if this actually  is what this results in. Will double check later, and send you a screenshot.

Reply:
![grafik](https://user-images.githubusercontent.com/11978847/116718076-e599ee80-a9d9-11eb-8e04-d7d8a204b9a5.png)

The main thing that need's to be improved now is just thickness, through a better rasterization. 

Would it make sense to look into the fontrenderer, and maybe delegate this to it as well?

Reply:
We can't pass this sort of data into freetype, at least, not without defining a glyph in a font file.
I think this looks good.

I'm going to merge this, but adjust this line to be `color = under_color * o_underline_color;` as I don't think the more complex math is required; the prior color is the background color and it's fine to just stamp the underline color over the top and make the shader a bit cheaper to execute.